Build with clangStdenv (`libui.override { stdenv = clangStdenv; }`) still fails:
```
/build/source/unix/text.c:16:9: error: call to undeclared function 'strcasecmp'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
   16 |         return strcasecmp(a, b);
      |                ^
/build/source/unix/text.c:16:9: note: did you mean 'g_strcasecmp'?
/nix/store/7r91q2lz5c91wk2f04a6kfswvfbngfmw-glib-2.82.4-dev/include/glib-2.0/glib/gstrfuncs.h:268:23: note: 'g_strcasecmp' declared here
  268 | gint                  g_strcasecmp     (const gchar *s1,
      |                       ^
1 error generated.
```

--- a/pkgs/development/libraries/libui/default.nix
+++ b/pkgs/development/libraries/libui/default.nix
@@ -30,6 +30,11 @@ stdenv.mkDerivation rec {
     lib.optional stdenv.hostPlatform.isLinux gtk3
     ++ lib.optionals stdenv.hostPlatform.isDarwin [ Cocoa ];
 
+  postPatch = ''
+    substituteInPlace darwin/text.m unix/text.c \
+      --replace-fail "strcasecmp" "g_strcasecmp"
+  '';
+
   preConfigure = lib.optionalString stdenv.hostPlatform.isDarwin ''
     sed -i 's/set(CMAKE_OSX_DEPLOYMENT_TARGET "10.8")//' ./CMakeLists.txt
   '';

